packages/react/src/use-pagination/index.ts

```diff
@@ -25,7 +25,7 @@ const usePagination = ({
   const [activePage, setActivePage] = useState(page || initialPage);
 
   const onChangeActivePage = (newPage: number) => {
-    setActivePage(newPage);
+    setActivePage(page ?? newPage);
```

Comment:
Hey @CNLHC thanks for sending this PR, could you remove this?, instead add these lines:

```
import { useMemo, useCallback, useState, useEffect } from 'react';
....
  useEffect(() => {
    if (page && page !== activePage) {
      setActivePage(page);
    }
  }, [page]);
  ...
```
Because otherwise, the `activePage` will be out of date with respect to `page`

Reply:
Thanks for your review. I think it is necessary to add a `useEffect` to set `activePage` when the `page` changes. 

But I think the `setActivePage(page ?? newPage);`  is also useful. 

For instance, if a user clicks the next button in a controlled `Pagination`, without this statement, the `activePage` will differ from the `page`. The expected way is to keep `page` and `activePage` synchronization and notify the `newPage` value using `onChange`, the parent component should provide an `onChange` callback that takes care of the state updating. Once the upper state is updated, the `useEffect` will  update the `activePage` to the `newPage`

Reply:
Hey @CNLHC thank you for helping the project, I understand your point but the `activePage` won't defer from the `page` because `next` and `previous` methods use the `activePage` (which is synchronized in the `useEffect` with the `page` value) for changing the current page, so if the user wants to update its `page` has to listen to `onChange(newPage)` event, to sum up since the `useEffect` is updating the `activePage` when `page` changes, is not necessary to change it again in the `onChangeActivePage` method